<?xml version="1.0"?>
<launch>
    
<!--    <include file="$(find nao_maccs)/launch/nao_control.launch"/> -->
    
  <!-- Load the URDF Model -->
  <include file="$(find nao_description)/launch/upload_nao.launch" />  
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
<!--     <arg name="world_name" value="$(find nao_gazebo_plugin)/worlds/nao_test.world"/> -->
<!--     <arg name="paused" value="true"/> -->
  </include> 


  <!-- Use PIDs?? -->

  <arg name="use_pid" value="true"/>
  <group if="$(arg use_pid)">
    <rosparam file="$(find nao_gazebo_plugin)/config/gazebo_ros_control_params.yaml" command="load"/>
  </group>

  <!-- Call Nao Robot Trajectory Controller -->

  <include file="$(find nao_control)/launch/nao_control_trajectory.launch"/>
<!--
  <include file="$(find nao_control)/launch/nao_control_position.launch"/>
-->
  <!-- Spawn a robot into Gazebo -->
 <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-param robot_description -urdf -x 1 -y 1 -z 0.335 -model nao_robot_v4_H25" />
  
 
 <!-- Set configuration of the pattern generator --> 
    <param name="mogs_soth_pg/robot_urdf" value="$(find ros-mogs-soth-pg)/mogs/nao.urdf"/>
<!--     <param name="mogs_soth_pg/prefix" value=""/> -->
    <param name="mogs_soth_pg/config" value="$(find ros-mogs-soth-pg)/config/nao.xml"/>
  
    <node name="mogs_soth_pg_node" pkg="ros-mogs-soth-pg" type="mogs_soth_pg_node">
        <remap from="/mogs_soth_pg/joint_states" to="/joint_states" />
        <remap from="/mogs_soth_pg/joint_angles" to="/joint_angles" />
    </node>
    
    <node name="angle_traj_to_nao_gazebo" pkg="ros-mogs-soth-pg" type="angle_traj_to_nao_gazebo"/>
  
</launch>
